File: backend/internal/domain/item/item.go
package item

import "time"

type Unit string

const (
	UnitGrams Unit = "grams"
	UnitML    Unit = "ml"
	UnitL     Unit = "l"
)

type Packaging string

const (
	PackagingBottle  Packaging = "bottle"
	PackagingCan     Packaging = "can"
	PackagingBox     Packaging = "box"
	PackagingBag     Packaging = "bag"
	PackagingJar     Packaging = "jar"
	PackagingPackage Packaging = "package"
	PackagingOther   Packaging = "other"
)

type Item struct {
	ID            string
	InventoryTag  string
	Name          string
	BestBefore    time.Time
	ContentAmount float64
	ContentUnit   Unit
	Packaging     Packaging
	PictureKey    *string
	Comment       *string
	CreatedAt     time.Time
	UpdatedAt     time.Time
	DeletedAt     *time.Time
}

File: frontend/src/app/core/models/item.ts
export type ContentUnit = 'grams' | 'ml' | 'l';
export type Packaging = 'bottle' | 'can' | 'box' | 'bag' | 'jar' | 'package' | 'other';

export interface Item {
  id: string;
  inventoryTag: string;
  name: string;
  bestBefore: string;
  contentAmount: number;
  contentUnit: ContentUnit;
  packaging: Packaging;
  pictureKey: string | null;
  comment?: string;
  createdAt: string;
  updatedAt: string;
}

export interface JsonApiItemAttributes {
  name: string;
  best_before: string;
  content_amount: number;
  content_unit: ContentUnit;
  packaging: Packaging;
  picture_key: string | null;
  comment?: string;
  inventory_tag: string;
  created_at: string;
  updated_at: string;
}

export interface JsonApiItemResource {
  type: 'items';
  id: string;
  attributes: JsonApiItemAttributes;
}

export interface JsonApiListResponse {
  data: JsonApiItemResource[];
}

export type ItemSortBy = 'name' | 'best_before' | 'created_at' | 'updated_at';
export type SortOrder = 'asc' | 'desc';

export interface CreateItemInput {
  name: string;
  bestBefore: string;
  contentAmount: number;
  contentUnit: ContentUnit;
  packaging: Packaging;
  pictureKey: string | null;
  comment?: string;
}
